android/java/org/chromium/chrome/browser/settings/BraveNewsCategorySources.java

```diff
@@ -0,0 +1,192 @@
+        categoryArg = args.getString("category");
+    }
+
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        InitBraveNewsController();
+        addPreferencesFromResource(R.xml.brave_news_sources_default);
+        getActivity().setTitle(categoryArg);
+
+        categsPublishers = new TreeMap<>();
+        preferenceManager = getPreferenceManager();
+        mBraveNewsController.getPublishers((publishers) -> {
+            List<Publisher> allPublishers = new ArrayList<>();
+            List<Publisher> categoryPublishers = new ArrayList<>();
+            for (Map.Entry<String, Publisher> entry : publishers.entrySet()) {
+                String key = entry.getKey();
+
+                Publisher publisher = entry.getValue();
+                if (publisher.categoryName.toLowerCase(Locale.ROOT)
+                                .equals(categoryArg.toLowerCase(Locale.ROOT))) {
+                    // categsPublishers.computeIfAbsent(publisher.categoryName, k ->new
+                    // ArrayList<>()).add(publisher);
+                    categoryPublishers.add(publisher);
+                }
+
+                categsPublishers.put(publisher.categoryName, categoryPublishers);
+                allPublishers.add(publisher);
+            }
+            Comparator<Publisher> compareByName =
+                    (Publisher o1, Publisher o2) -> o1.publisherName.compareTo(o2.publisherName);
+            Collections.sort(categoryPublishers, compareByName);
+            Collections.sort(allPublishers, compareByName);
+            categsPublishers.put("All Sources", allPublishers);
+
+            addCategs(categsPublishers);
+        });
+    }
+
+    @Override
+    public void onConnectionError(MojoException e) {
+        if (mBraveNewsController != null) {
+            mBraveNewsController.close();
+        }
+        mBraveNewsController = null;
+        InitBraveNewsController();
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        if (mBraveNewsController != null) {
+            mBraveNewsController.close();
+        }
+    }
+
+    private void InitBraveNewsController() {
+        if (mBraveNewsController != null) {
+            return;
+        }
+
+        mBraveNewsController =
+                BraveNewsControllerFactory.getInstance().getBraveNewsController(this);
+    }
+
+    private void addCategs(TreeMap<String, List<Publisher>> publisherCategories) {
+        PreferenceScreen sourcesScreen = getPreferenceScreen();
+
+        SearchPreference search = new SearchPreference(ContextUtils.getApplicationContext());
+        search.setKey("search_pref");
+
+        sourcesScreen.addPreference(search);
+
+        for (Map.Entry<String, List<Publisher>> map : publisherCategories.entrySet()) {
+            String category = map.getKey();
+            if (category.toLowerCase(Locale.ROOT).equals(categoryArg.toLowerCase(Locale.ROOT))) {
```

Comment:
why `for` loop is used to get the value from the map by the key instead of something like
```
List<Publisher> publishers = map.find(categoryArg);
```

Reply:
not sure i'm familiar with that. looks like i needed the for loop to match the category set from the bundle with  the one from the api response.

Reply:
I meant method `get` https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-

Reply:
refactored [here](https://github.com/brave/brave-core/blob/ffc56b02ed22ecd9e5bbc2769806f28bd364e821/android/java/org/chromium/chrome/browser/settings/BraveNewsCategorySources.java#L125)
the `toLowerCase` should not be necessary . before i used it to be sure when comparing but the lower case values were not used otherwise.

Reply:
thanks